R/utils.R

```diff
@@ -16,8 +16,10 @@ check_factor <- function(f) {
     factor(f)
   } else if (is.factor(f)) {
     f
+  } else if (is.numeric(f)) {
```

Comment:
This seems a little over broad to me - I think it's ok to permit explicit coercion, but not to do it automatically (as that makes it easier for mistakes to slip through)

Reply:
I'm happy to make this change, of course, but for learning purposes, can I ask about the philosophy here?

If someone passes a character vector to `fct_inorder()`, `fct_count()`, or similar, we assume they intended to factor that vector, and we coerce it in `check_factor()`.

If someone passes a numeric vector for `fct_inorder()`, why should we assume this was a mistake rather than auto-coercing?

Reply:
In an ideal world, I don't think `fct_inorder()` etc would accept either character or factors, but because they're used almost interchangeably in R, I think it would be too frustrating if these functions didn't accept either. But numeric vectors are quite different from factors, so I think it makes sense to require the user to explicit coerce.  It's basically a tradeoff: it makes life a little bit harder sometimes in return for getting clearer error messages when you've actually made a mistake.